File: frontend/ddn-frontend/src/app/api/auth/auth.api.ts
import { inject, Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { API_BASE_URL } from '../../app.config';

export type LoginRequest = {
  email: string;
  password: string;
};

export type LoginResponse = {
  token: string;
};

@Injectable({ providedIn: 'root' })
export class AuthApi {
  private http = inject(HttpClient);
  private baseUrl = inject(API_BASE_URL); // npr. http://localhost:8080/api

  login(req: LoginRequest): Observable<LoginResponse> {
    return this.http.post<LoginResponse>(`${this.baseUrl}/auth/login`, req);
  }
}

File: android/TaxiMobile/app/src/main/java/com/example/taximobile/feature/auth/data/AuthApi.java
package com.example.taximobile.feature.auth.data;

import com.example.taximobile.feature.auth.data.dto.request.LoginRequest;
import com.example.taximobile.feature.auth.data.dto.response.LoginResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface AuthApi {

    @POST("api/auth/login")
    Call<LoginResponse> login(@Body LoginRequest req);
}
